### ISSUE TITLE
Assertion failed: infeasible in HiGHS

### ISSUE THREAD

**Issue by Thell** [ERGO-Code/HiGHS#1267]
```
--->8---
(state_1_sum_lb, state_2_sum_lb): (952, 3)
(state_1_sum_lb, state_2_sum_lb): (927, 29)
(state_1_sum_lb, state_2_sum_lb): (149, 45)
Assertion failed: infeasible_ == 0, file C:\Users\thell\.cargo\git\checkouts\highs-sys-libz-228d8310235e4834\fdbc79a\HiGHS\src\mip/HighsDomain.h, line 483 
```
https://github.com/ERGO-Code/HiGHS/blob/4978e1e06a29d2682a1226c75356cd52359a52e2/src/mip/HighsDomain.h#L480

This is in the `latest` branch and is breaking in both debug and release when executing Rust code built with https://github.com/rust-or/highs-sys so perhaps I need to change a runtime flag or perhaps @lovasoa needs to pass something to CMake or, lastly, perhaps an INFEASIBLE should be an assert? I don't know...

I'm processing all (state_1_sum_lb, state_2_sum_lb) pairs from (0,0) to (1116, 46) in random order. I expect numerous `Infeasible` but I don't expect any runtime assertions. All of the iterations complete successfully using pywraplp with the CBC backend.

I read in the model, change the rows bounds for the rows associated with the state value sum bounds ([270, 271] in this model) and then if the result is `MODEL_STATUS_OPTIMAL` I extract the solution and if it isn't I ignore it.

[subset_selection.txt](https://github.com/ERGO-Code/HiGHS/files/11291103/subset_selection.txt)

So, in short... if it is INFEASIBLE that shouldn't cause a runtime assertion to fail, but if it is a true model _error_ then yeah, I'd expect a failure.

**Comment by Thell** [comment]
It looks like this is my error. I had updated the submodule to the 'latest' branch yet had not committed the updated submodule revision so I was still using a revision that had a naked assertion instead of the `d4809eae7` protected one.

Lesson learned: check the src in the build directory even if everything looks right in your other dependency modules.

Sorry about wasting the time of anyone who looked at this.

**Comment by jajhall** [comment]
Glad you've got it sorted. We'd not started to look from our end 🙂

**Comment by Thell** [comment]
Well, it's kinda funny... within minutes of getting that fixed and running through more of my tests sets I hit:

```
        // ToDo Eliminate setBasisValidity once ctest passes. Asserts
        // verify that it does nothing - other than setting
        // info_.valid = true;
        setBasisValidity();
        assert(model_status_ == HighsModelStatus::kInfeasible ||
               model_status_ == HighsModelStatus::kUnbounded);
        return returnFromRun(return_status);
```

so... this is getting to be an adventure I didn't plan on having.
